### PR TITLE
WIP: Add status condition types

### PR DESCRIPTION
**Pull request by mrogers950**
Just sketching out the types for now.

### CODE REVIEW

File: pkg/apis/fileintegrity/v1alpha1/fileintegrity_types.go [hunk 1]
```diff
@@ -15,6 +17,13 @@ const (
 	PhasePending      FileIntegrityStatusPhase = "Pending"
 )
 
+type FileIntegrityNodeCondition string
+
+const (
+	NodeConditionGood FileIntegrityNodeCondition = "Good"
+	NodeConditionBad  FileIntegrityNodeCondition = "Bad"
```
Comment (JAORMX) [inline pkg/apis/fileintegrity/v1alpha1/fileintegrity_types.go]: what about succeed/fail ?

"the integrity check succeeded/failed"

File: pkg/apis/fileintegrity/v1alpha1/fileintegrity_types.go [hunk 2]
```diff
@@ -38,7 +47,20 @@ type FileIntegrityStatus struct {
...
+}
+
+type FileIntegrityStatusConditions struct {
+	Nodes []NodeStatus `json:"nodes"`
+}
+
+type NodeStatus struct {
+	NodeName                 string                     `json:"nodeName"`
+	LastProbeTime            time.Time                  `json:"lastProbeTime"`
+	Condition                FileIntegrityNodeCondition `json:"condition"`
```
Comment (JAORMX) [inline pkg/apis/fileintegrity/v1alpha1/fileintegrity_types.go]: how would this looks like?

File: pkg/apis/fileintegrity/v1alpha1/fileintegrity_types.go (hunk 1 above)
Comment (mrogers950) [reply pkg/apis/fileintegrity/v1alpha1/fileintegrity_types.go]: Succeeded / Failed works for me

File: pkg/apis/fileintegrity/v1alpha1/fileintegrity_types.go (hunk 2 above)
Comment (mrogers950) [reply pkg/apis/fileintegrity/v1alpha1/fileintegrity_types.go]: The CR status would look something like:
```
status:
  phase: Active
  nodeStatus:
  - nodeName: foo
    lastProbeTime: <timestamp>
    condition: Succeeded
  - nodeName: foo2
    lastProbeTime: <timestamp>
    condition: Failed
    resultConfigMapName: foo2-aide-results-asdf
    resultConfigMapNamespace: openshift-file-integrity

**Comment by JAORMX** [comment]
Yep.... this is broken as was the compliance-operator:

https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_file-integrity-operator/15/pull-ci-openshift-file-integrity-operator-master-e2e-aws/36

the issue is:

```
 go test: mod flag may be set only once 
```

We need to merge: https://github.com/openshift/file-integrity-operator/pull/23

File: pkg/apis/fileintegrity/v1alpha1/fileintegrity_types.go (hunk 2 above)
Comment (JAORMX) [reply pkg/apis/fileintegrity/v1alpha1/fileintegrity_types.go]: I dig that!

**Comment by JAORMX** [comment]
OK, now we're getting better results for this. There are genuine errors in the code.

**Comment by mrogers950** [comment]
Updated the design doc with some revisions.

**Pull request closed without merging**